x-pack/plugins/apm/public/components/app/service_inventory/index.tsx

```diff
@@ -280,19 +295,74 @@ export function ServiceInventory() {
     'serviceName'
   );
 
+  const mlCallout = (
+    <EuiFlexItem>
+      <MLCallout
+        isOnSettingsPage={false}
+        anomalyDetectionSetupState={anomalyDetectionSetupState}
+        onDismiss={() => setUserHasDismissedCallout(true)}
+      />
+    </EuiFlexItem>
+  );
+
+  const kibanaUiServiceLimitCallout = (
+    <EuiFlexItem>
+      <EuiCallOut
+        title={i18n.translate(
+          'xpack.apm.serviceList.ui.limit.warning.calloutTitle',
+          {
+            defaultMessage:
+              'The performance limit to the number of services reached',
+          }
+        )}
+        color="danger"
+        iconType="alert"
+      >
+        <EuiText size="s">
+          <FormattedMessage
+            defaultMessage="The number of unique service names limit was exceeded. Try narrowing down your results using the search bar"
+            id="xpack.apm.serviceList.ui.limit.warning.calloutDescription"
+          />
+        </EuiText>
+      </EuiCallOut>
+    </EuiFlexItem>
+  );
+
+  const maxGroupReachedCallout = (
+    <EuiFlexItem>
+      <EuiCallOut
+        title={i18n.translate(
+          'xpack.apm.serviceList.apmServer.limit.warning.calloutTitle',
+          {
+            defaultMessage: 'The limit to the number of services reached',
+          }
+        )}
+        color="danger"
+        iconType="alert"
+      >
+        <EuiText size="s">
+          <FormattedMessage
+            defaultMessage="The maximum number of unique services has been reached. Please increase {codeBlock} in APM Server."
+            id="xpack.apm.serviceList.apmServer.limit.warning.calloutDescription"
+            values={{
+              codeBlock: <EuiCode>aggregation.service.max_groups</EuiCode>,
+            }}
+          />
+        </EuiText>
+      </EuiCallOut>
+    </EuiFlexItem>
+  );
+
   return (
     <>
       <SearchBar showTimeComparison />
       <EuiFlexGroup direction="column" gutterSize="m">
-        {displayMlCallout && (
-          <EuiFlexItem>
-            <MLCallout
-              isOnSettingsPage={false}
-              anomalyDetectionSetupState={anomalyDetectionSetupState}
-              onDismiss={() => setUserHasDismissedCallout(true)}
-            />
-          </EuiFlexItem>
-        )}
+        {displayMlCallout && mlCallout}
+        {hasKibanaUiLimitRestrictedData
+          ? kibanaUiServiceLimitCallout
+          : shouldDisplayMaxGroupReachedCallout
+          ? maxGroupReachedCallout
+          : null}
```

Comment:
I have defined individual callouts as components on the top and used this ternary behaviour to control the display.
This way the JSX part looks a bit clean and readable. But let me know if you otherwise.

To explain the logic, the requirement states - `We must prioritise Kibana UI Limit over APM Server Limit when displaying the callout`